packages/backend/scripts/diffc.py

```diff
@@ -0,0 +1,241 @@
+        sol_files2 = [os.path.join(root, f) for root, dirs, files in os.walk(
+            dir_path2) for f in files if f.endswith('.sol')]
+
+        # Map to common format
+        sol_files1 = {f.replace(dir_path1, "<placeholder>")
+                      for f in sol_files1}
+        sol_files2 = {f.replace(dir_path2, "<placeholder>")
+                      for f in sol_files2}
+
+        common_sol_files = sol_files1 & sol_files2
+
+        for sol_file in common_sol_files:
+            file1 = sol_file.replace(
+                "<placeholder>", dir_path1).replace(folder1, folder1)
+            file2 = sol_file.replace(
+                "<placeholder>", dir_path2).replace(folder1, folder2)
+
+            # Run "forge fmt" on each file before diffing
+            # subprocess.run(["forge", "fmt", file1])
+            # subprocess.run(["forge", "fmt", file2])
+
+            pid, fd = pty.fork()
+            if pid == 0:  # child process
+                os.execvp("difft", ["difft", "--skip-unchanged",
+                          "--ignore-comments", "--width", str(terminal_width), file1, file2])
+            else:  # parent process
+                result = b""
+                while True:
+                    try:
+                        chunk = os.read(fd, 1024)
+                        if not chunk:  # no more data
+                            break
+                        result += chunk
+                    except OSError:
+                        break
+                result = result.decode()
+                if "No changes." not in result:
+                    # Split result into lines
+                    lines = result.split('\n')
+                    # Only print lines that are not empty
+                    for line in lines:
+                        if line.strip():  # This checks if line is not empty when whitespace is removed
+                            print(line)
+                    if (line not in [""]):
+                        no_changes = False
+
+        if no_changes:
+            print("No changes in implementations.")
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="List directories within a specified directory and compare proxies.")
+
+    parser.add_argument("Folder_Name1", metavar="folder_name1",
+                        type=str, help="the first folder name to list directories")
+    parser.add_argument("Folder_Name2", metavar="folder_name2",
+                        type=str, help="the second folder name to list directories")
+
+    args = parser.parse_args()
+
+    directories1, ignored1 = list_directories(args.Folder_Name1)
+    directories2, ignored2 = list_directories(args.Folder_Name2)
+
+    common_directories = directories1 & directories2
+    unique_to_folder1 = directories1 - directories2
+    unique_to_folder2 = directories2 - directories1
+    ignored_directories = ignored1 | ignored2
+
+    if common_directories:
```

Comment:
What do you think about printing something like this:

```
Common Directories: ....
Directories unique to 1: NONE
Directories unique to 2: ......
Ignored directories: NONE
```

So we would print `NONE` when there are no directories for a given arrangement. That way we always print something about all four of these values and the possible confusion "Where are x?" disappear

Reply:
done